### PR TITLE
SW-4767 Implement the project featured filter in the nursery withdrawals view

### PR DESCRIPTION
**Pull request by nickgraz**
SW-4767 Implement the project featured filter in the nursery withdrawals view

Move the pill renderer to the featured filter config

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2130?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #2102** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2102?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2103** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2103?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #2117** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2117?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #2128** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2128?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #2129** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2129?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #2130** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2130?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2130?utm_source=stack-comment).

File: src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx [hunk 1]
```diff
@@ -86,6 +103,38 @@ export default function NurseryWithdrawalsTable(): JSX.Element {
...
+                operation: 'field',
+                type: 'Exact',
+                values: values.map((value: number | string | null): string | null =>
+                  value === null ? value : `${value}`
+                ),
+              }),
+              label: strings.PROJECTS,
+              renderOption: (id: string | number) => getProjectName(Number(id)),
+              notPresentFilterShown: true,
+              notPresentFilterLabel: activeLocale ? strings.NO_PROJECT : '',
```
Comment (karthikbtf) [inline src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx:124]: gate the entire payload with featureFlagProjects and activeLocale

File: src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx [hunk 2]
```diff
@@ -86,6 +103,38 @@ export default function NurseryWithdrawalsTable(): JSX.Element {
     [filterColumns, filterOptions, filters]
   );
 
+  const featuredFilters: FeaturedFilterConfig[] = useMemo(
```
Comment (karthikbtf) [inline src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx:106]: thoughts on a reusable hook to populate this (similar to what's in planting progress).. only if it makes sense to do so

**Review by karthikbtf — APPROVED** [review]
Thanks for adding types in the selectors/responses.

File: src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx (hunk 2 above)
Comment (nickgraz) [reply src/scenes/NurseryRouter/NurseryWithdrawalsTable.tsx:106]: Good call, the only difference between the two current implementations is the `field` within the `searchNodeCreator`. I'll refactor it in a follow up PR

**Review by karthikbtf — APPROVED** [review]
Thanks for the cleanup!!

**Comment by nickgraz** [comment]
### Merge activity

* **Feb 7, 1:21 PM EST**: @nickgraz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2130).

**Pull request merged**